### ISSUE TITLE
Removing Items and Updating Quantities in Mini-Cart block (header) have issues

### ISSUE THREAD

**Issue by ADDISON74** [nexcess/magento-turpentine#1034]
Clear you Magento cache and session folder, clear your browser cache and cookies, restart Varnish service.

1) Add any product to cart. Once you add it to cart you will be redirect to cart page. See how mini-cart block in header is updating its content with the product you just added. Click on mini-cart then click on [Remove Item] button.

![remove11](https://cloud.githubusercontent.com/assets/8360474/11587704/e3b214be-9a83-11e5-9126-c3a45b1cfd57.jpg)

2.1) Product is removed from mini-cart block but cart content is not updating. Product is still there. If you click on update [UPDATE SHOPPING CART] button you will get the page content "You have no items in your shopping cart. Click here to continue shopping". So, removing item from mini-cart block removes the product from shopping cart.

![remove12](https://cloud.githubusercontent.com/assets/8360474/11587773/93a29650-9a84-11e5-8ef2-38943febf62e.jpg)

3) Please note also other issue when you are in cart page if you Update the quantity in mini-cart block it won't update in cart page. But if you update the quantity in cart page, quantity in mini-cart block will be updated accordingly.

In conclusion mini-cart block have issue two issue: "Removing Items" and "Update Quantity".

**Comment by aricwatson** [comment]
From my testing, this is actually the *same* behavior as regular Magento. You can verify this by testing on a non-Varnish install of the Magento, or just turn on the Varnish bypass cookie and testing.

Maybe not good UX, but the same as the default install, which is what we're shooting for :)

**Comment by ADDISON74** [comment]
You are right. I checked without Varnish and it is a Magento issue only for the cart page. That remove link is hard coded, it is not in a template file. The only solution getting this working is editing a core file deleting the part which adds the encoded link. I will close it and post in Magento bugtracker.

PS - I will open a issue with more detail related to uenc links and form_keys. It seems Varnish caches this links and that's why removing items from compare side block is not working or there are strange redirects.
